apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/triggers_handler.ex

```diff
@@ -557,22 +575,22 @@ defmodule Astarte.DataUpdaterPlant.TriggersHandler do
         retry
       end
 
-    publish(exchange, routing_key, payload, opts)
-    |> wait_backoff_and_publish(next_retry, exchange, routing_key, payload, opts)
+    publish(server, exchange, routing_key, payload, opts)
+    |> wait_backoff_and_publish(server, next_retry, exchange, routing_key, payload, opts)
   end
 
-  defp wait_ok_publish(exchange, routing_key, payload, opts) do
-    publish(exchange, routing_key, payload, opts)
-    |> wait_backoff_and_publish(1, nil, routing_key, payload, opts)
+  defp wait_ok_publish(server, exchange, routing_key, payload, opts) do
+    publish(server, exchange, routing_key, payload, opts)
+    |> wait_backoff_and_publish(server, 1, nil, routing_key, payload, opts)
   end
 
-  defp publish(nil, routing_key, payload, opts) do
+  defp publish(_server, nil, routing_key, payload, opts) do
     Config.events_exchange_name!()
     |> AMQPEventsProducer.publish(routing_key, payload, opts)
   end
```

Comment:
I think `nil` here only means we use the default exchange, not that we're discarding the value

```suggestion
  defp publish(server, nil, routing_key, payload, opts) do
    exchange = Config.events_exchange_name!()
    AMQPEventsProducer.publish(server, exchange, routing_key, payload, opts)
  end
```

I believe it would be better to just join the two `publish` bodies with

```elixir
  defp publish(server, exchange, routing_key, payload, opts) do
    exchange = exchange || Config.events_exchange_name!()
    AMQPEventsProducer.publish(server, exchange, routing_key, payload, opts)
  end
```